### DISCUSSION TITLE
QLocalServer::listen: Name error with long BASEDIR

### DISCUSSION THREAD

**Discussion by Stas1981ru** [qutebrowser/qutebrowser#5697] [Q&A] [answered]
When I set BASEDIR with small count of char , like "/xxx/xxx/xxx/xxx" is ok.
When the length of BASEDIR is mach longer, like "/xxx/xxx/.............../xxx" qutebrowser give error "qutebrowser.misc.ipc.ListenError: Error while listening to IPC server: QLocalServer::listen: Name error (error 2)"

The setter of  the name is the function misc.ipc._get_socketname(basedir) , which return:
"return os.path.join(standarddir.runtime(), filename)"

I rewrite this to: "return os.path.join(filename)" the variable "filename" contains md5 of BASEDIR

Will this "hack" have no further consequences ?

**Comment by toofar** [comment, marked as answer]
Would that make it relative to whatever directory qutebrowser was started from? If so it might make opening URLs in an existing instance break. Maybe `os.path.join(QStandardPaths.writableLocation(QStandardPaths.RuntimeLocation), filename)` would put it in the XDG runtime dir with the basedir hash in the filename.

**Reply by Stas1981ru** [reply]
>Would that make it relative to whatever directory qutebrowser was started from?
No, all instance of qutebroweser will be start with new session from separated directories with -R command line options.
Thanks.
